CcrRetentionLeaseIT#testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes is flaky

This test looks like it fails rarely (I see one failure), I found it while looking in the build stats for another test case. The failure is this:

```
java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
	at __randomizedtesting.SeedInfo.seed([375ED352592087EB:C0F2CA3AA4080F01]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:858)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:832)
	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:360)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: <1550855178372L>
     but: was <1550855178575L>
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$5(CcrRetentionLeaseIT.java:376)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:846)
		... 39 more
```

I understand the reason for the failure and will push a fix shortly.

I'm re-opening because this test has failed several times over the past couple days. I'm not able to reproduce it locally. The assertion error is a bit different from the one above, but it seemed related enough that I avoided filing another issue.

---

Link to build logs: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=openjdk12,nodes=immutable&&linux&&docker/125/console

Command to reproduce:
```
./gradlew :x-pack:plugin:ccr:internalClusterTest \
  -Dtests.seed=DD1D8CB425936EBD \
  -Dtests.class=org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT \
  -Dtests.method="testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" \
  -Dtests.security.manager=true \
  -Dtests.locale=nyn \
  -Dtests.timezone=Asia/Famagusta \
  -Dcompiler.java=12 \
  -Druntime.java=12
```

Relevant excerpt from logs:
```
FAILURE 19.7s J4 | CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: a collection with size <1>
   >      but: collection size was <0>
   > 	at __randomizedtesting.SeedInfo.seed([DD1D8CB425936EBD:2AB195DCD8BBE657]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:315)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:862)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:306)
```

Another failure of this in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3174/console with:

```
org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT > testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes FAILED
    java.lang.AssertionError: 
    Expected: a collection with size <1>
         but: collection size was <0>
        at __randomizedtesting.SeedInfo.seed([17606BB64105D4F0:E0CC72DEBC2D5C1A]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:316)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:862)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
        at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:307)
REPRODUCE WITH: ./gradlew :x-pack:plugin:ccr:internalClusterTest --tests "org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" -Dtests.seed=17606BB64105D4F0 -Dtests.security.manager=true -Dtests.locale=lrc-IR -Dtests.timezone=Africa/El_Aaiun -Dcompiler.java=12 -Druntime.java=11
```

We have another failure in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java11,nodes=immutable&&linux&&docker/154/console:

reproduction line (does not reproduce locally):

```
./gradlew :x-pack:plugin:ccr:internalClusterTest --tests "org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" \
  -Dtests.seed=1FB0574A684FE4CE \
  -Dtests.security.manager=true \
  -Dtests.locale=ks-IN \
  -Dtests.timezone=Asia/Samarkand \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

error message:

```
java.lang.AssertionError: 
Expected: a collection with size <1>
     but: collection size was <0>
	at __randomizedtesting.SeedInfo.seed([1FB0574A684FE4CE:E81C4E2295676C24]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:316)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:868)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:842)
	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:307)
[...]
```

Attached are the [full test logs](https://github.com/elastic/elasticsearch/files/3148382/full-test-log.txt).

Due to https://github.com/elastic/elasticsearch/issues/39331#issuecomment-484426987 I am intentionally *not* muting this test on that branch.

And one more failure here: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java11,nodes=immutable&&linux&&docker/12/console

reproduction line:
```
./gradlew :x-pack:plugin:ccr:internalClusterTest \
  -Dtests.seed=C56ECDFBA213A5C5 \
  -Dtests.class=org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT \
  -Dtests.method="testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" \
  -Dtests.security.manager=true \
  -Dtests.locale=gv \
  -Dtests.timezone=Europe/Vaduz \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

error message:
```
  > Throwable #1: java.lang.AssertionError: 
   > Expected: a collection with size <1>
   >      but: collection size was <0>
   > 	at __randomizedtesting.SeedInfo.seed([C56ECDFBA213A5C5:32C2D4935F3B2D2F]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:316)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:858)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:832)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:307)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

And one more: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java12,nodes=immutable&&linux&&docker/12/console

reproduction line:
```
./gradlew :x-pack:plugin:ccr:internalClusterTest \
  -Dtests.seed=CD37DCDD0A66F5D5 \
  -Dtests.class=org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT \
  -Dtests.method="testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" \
  -Dtests.security.manager=true \
  -Dtests.locale=ee \
  -Dtests.timezone=Asia/Krasnoyarsk \
  -Dcompiler.java=12 \
  -Druntime.java=12
```

error message: 
```
   > Throwable #1: java.lang.AssertionError: 
   > Expected: a collection with size <1>
   >      but: collection size was <0>
   > 	at __randomizedtesting.SeedInfo.seed([CD37DCDD0A66F5D5:3A9BC5B5F74E7D3F]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:316)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:858)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:832)
   > 	at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:307)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
   > 	at java.base/java.lang.Thread.run(Thread.java:835)
```

Here are the [full logs](https://github.com/elastic/elasticsearch/files/3153396/CcrRetentionLeaseIT__testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes.2.txt.zip).

These logs do not make any sense:

```
[2019-05-06T19:02:05,894][INFO ][o.e.x.c.CcrRetentionLeaseIT] [testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes] before test
[2019-05-06T19:02:05,902][INFO ][o.e.c.m.MetaDataCreateIndexService] [leaderm0] [leader] creating index, cause [api], templates [], shards [2]/[0], mappings [doc]
[2019-05-06T19:02:06,141][INFO ][o.e.c.r.a.AllocationService] [leaderm0] Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[leader][1]] ...]).
[2019-05-06T19:02:06,188][INFO ][o.e.x.c.CcrRetentionLeaseIT] [testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes] ensure green leader indices [leader]
[2019-05-06T19:02:06,189][INFO ][o.e.x.c.CcrRetentionLeaseIT] [testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes] indexing [1429] docs
[2019-05-06T19:02:10,247][WARN ][o.e.x.c.r.CcrRepository  ] [follower1] [follower][1] background renewal of retention lease [follower_cluster/follower/6LUryH68TjK6oo7FYUk3Zw-following-leader_cluster/leader/SIJLPZqTTZavZS86igdjdw] failed during restore
org.elasticsearch.transport.RemoteTransportException: [leaderm0][127.0.0.1:36004][indices:admin/seq_no/renew_retention_lease]
Caused by: org.elasticsearch.transport.RemoteTransportException: [leaderd1][127.0.0.1:46101][indices:admin/seq_no/renew_retention_lease[s]]
Caused by: org.elasticsearch.index.seqno.RetentionLeaseNotFoundException: retention lease with ID [follower_cluster/follower/6LUryH68TjK6oo7FYUk3Zw-following-leader_cluster/leader/SIJLPZqTTZavZS86igdjdw] not found
```

This indicates that during the file copy phase, after we have *sync* taken a retention lease on the leader, we couldn't renew the lease because it did not exist. That does not make any sense at all. I will need more logging here to understand what is going on.

Additional failure:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=openjdk12,nodes=immutable&&linux&&docker/158/console

Reproduce line:
```
./gradlew :x-pack:plugin:ccr:internalClusterTest --tests "org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes" -Dtests.seed=1CD80A90A84F4656 -Dtests.security.manager=true -Dtests.locale=hsb-DE -Dtests.timezone=Europe/Lisbon -Dcompiler.java=12 -Druntime.java=12
```

```
08:41:29 org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT > testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes FAILED
08:41:29     java.lang.AssertionError: 
08:41:29     Expected: a collection with size <1>
08:41:29          but: collection size was <0>
08:41:29         at __randomizedtesting.SeedInfo.seed([1CD80A90A84F4656:EB7413F85567CEBC]:0)
08:41:29         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
08:41:29         at org.junit.Assert.assertThat(Assert.java:956)
08:41:29         at org.junit.Assert.assertThat(Assert.java:923)
08:41:29         at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.lambda$testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes$2(CcrRetentionLeaseIT.java:316)
08:41:29         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:868)
08:41:29         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:842)
08:41:29         at org.elasticsearch.xpack.ccr.CcrRetentionLeaseIT.testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes(CcrRetentionLeaseIT.java:307)
```

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRetentionLeaseIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRetentionLeaseIT.java
@@ -6,7 +6,6 @@
 
 package org.elasticsearch.xpack.ccr;
 
-import com.carrotsearch.hppc.cursors.ObjectCursor;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
@@ -44,7 +43,6 @@ import org.elasticsearch.indices.IndicesService;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.snapshots.RestoreInfo;
 import org.elasticsearch.snapshots.RestoreService;
-import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.transport.ConnectTransportException;
 import org.elasticsearch.transport.RemoteTransportException;
@@ -88,7 +86,6 @@ import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.hasSize;
 
-@TestLogging(value = "org.elasticsearch.xpack.ccr:trace,org.elasticsearch.indices.recovery:trace,org.elasticsearch.index.seqno:debug")
 public class CcrRetentionLeaseIT extends CcrIntegTestCase {
 
     public static final class RetentionLeaseRenewIntervalSettingPlugin extends Plugin {
@@ -224,9 +221,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
 
         // block the recovery from completing; this ensures the background sync is still running
         final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
-        for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
+        for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
             final MockTransportService senderTransportService =
-                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
             senderTransportService.addSendBehavior(
                     (connection, requestId, action, request, options) -> {
                         if (ClearCcrRestoreSessionAction.NAME.equals(action)
@@ -248,9 +245,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
             assertRetentionLeaseRenewal(numberOfShards, numberOfReplicas, followerIndex, leaderIndex);
             latch.countDown();
         } finally {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getDataNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.clearAllRules();
             }
         }
@@ -405,9 +402,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
 
         final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
         try {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.addSendBehavior(
                         (connection, requestId, action, request, options) -> {
                             if (RetentionLeaseActions.Remove.ACTION_NAME.equals(action)
@@ -456,9 +453,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
                 assertThat(Strings.toString(shardStats), shardStats.getRetentionLeaseStats().retentionLeases().leases(), empty());
             }
         } finally {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getDataNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.clearAllRules();
             }
         }
@@ -488,9 +485,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
 
         final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
         try {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.addSendBehavior(
                         (connection, requestId, action, request, options) -> {
                             if (RetentionLeaseActions.Remove.ACTION_NAME.equals(action)
@@ -526,9 +523,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
                             getLeaderCluster().getClusterName(),
                             new Index(leaderIndex, leaderUUID))));
         } finally {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getDataNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.clearAllRules();
             }
         }
@@ -766,35 +763,36 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
         final CountDownLatch latch = new CountDownLatch(1);
 
         final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
-        for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
-            final MockTransportService senderTransportService =
-                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
-            senderTransportService.addSendBehavior(
+        try {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
+                final MockTransportService senderTransportService =
+                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
+                senderTransportService.addSendBehavior(
                     (connection, requestId, action, request, options) -> {
                         if (RetentionLeaseActions.Renew.ACTION_NAME.equals(action)
-                                || TransportActionProxy.getProxyAction(RetentionLeaseActions.Renew.ACTION_NAME).equals(action)) {
+                            || TransportActionProxy.getProxyAction(RetentionLeaseActions.Renew.ACTION_NAME).equals(action)) {
                             senderTransportService.clearAllRules();
                             final RetentionLeaseActions.RenewRequest renewRequest = (RetentionLeaseActions.RenewRequest) request;
                             final String primaryShardNodeId =
-                                    getLeaderCluster()
-                                            .clusterService()
-                                            .state()
-                                            .routingTable()
-                                            .index(leaderIndex)
-                                            .shard(renewRequest.getShardId().id())
-                                            .primaryShard()
-                                            .currentNodeId();
+                                getLeaderCluster()
+                                    .clusterService()
+                                    .state()
+                                    .routingTable()
+                                    .index(leaderIndex)
+                                    .shard(renewRequest.getShardId().id())
+                                    .primaryShard()
+                                    .currentNodeId();
                             final String primaryShardNodeName =
-                                    getLeaderCluster().clusterService().state().nodes().get(primaryShardNodeId).getName();
+                                getLeaderCluster().clusterService().state().nodes().get(primaryShardNodeId).getName();
                             final IndexShard primary =
-                                    getLeaderCluster()
-                                            .getInstance(IndicesService.class, primaryShardNodeName)
-                                            .getShardOrNull(renewRequest.getShardId());
+                                getLeaderCluster()
+                                    .getInstance(IndicesService.class, primaryShardNodeName)
+                                    .getShardOrNull(renewRequest.getShardId());
                             final CountDownLatch innerLatch = new CountDownLatch(1);
                             // this forces the background renewal from following to face a retention lease not found exception
                             primary.removeRetentionLease(
-                                    getRetentionLeaseId(followerIndex, leaderIndex),
-                                    ActionListener.wrap(r -> innerLatch.countDown(), e -> fail(e.toString())));
+                                getRetentionLeaseId(followerIndex, leaderIndex),
+                                ActionListener.wrap(r -> innerLatch.countDown(), e -> fail(e.toString())));
 
                             try {
                                 innerLatch.await();
@@ -807,11 +805,18 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
                         }
                         connection.sendRequest(requestId, action, request, options);
                     });
+            }
+
+            latch.await();
+
+            assertRetentionLeaseRenewal(numberOfShards, numberOfReplicas, followerIndex, leaderIndex);
+        } finally {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
+                final MockTransportService senderTransportService =
+                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
+                senderTransportService.clearAllRules();
+            }
         }
-
-        latch.await();
-
-        assertRetentionLeaseRenewal(numberOfShards, numberOfReplicas, followerIndex, leaderIndex);
     }
 
     /**
@@ -858,9 +863,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
         final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
 
         try {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.addSendBehavior(
                         (connection, requestId, action, request, options) -> {
                             if (RetentionLeaseActions.Renew.ACTION_NAME.equals(action)
@@ -914,9 +919,9 @@ public class CcrRetentionLeaseIT extends CcrIntegTestCase {
                 assertThat(Strings.toString(shardStats), shardStats.getRetentionLeaseStats().retentionLeases().leases(), empty());
             }
         } finally {
-            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getDataNodes().values()) {
+            for (final DiscoveryNode senderNode : followerClusterState.getState().nodes()) {
                 final MockTransportService senderTransportService =
-                        (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.value.getName());
+                    (MockTransportService) getFollowerCluster().getInstance(TransportService.class, senderNode.getName());
                 senderTransportService.clearAllRules();
             }
         }

